pandas/tests/indexes/test_datetimelike.py

```diff
@@ -721,6 +721,17 @@ def test_fillna_datetime64(self):
                            dtype=object)
             self.assert_index_equal(idx.fillna('x'), exp)
 
+    def test_contains(self):
```

Comment:
further let's add testing with ``Timestamp`` and ``datetime`` as well (and a case where things are not there). finally add in a ``NaT`` both in the contained index as well as a tests with ``in``